### DISCUSSION TITLE
Defining same entity as multiple types

### DISCUSSION THREAD

**Discussion by mattkallo** [Netflix/dgs-framework#1558] [Q&A]
Let me try to explain my question: 

I have a table at the backend DB that stores 2 types of "entities" - eg  "Person" which stores both "Contact" and "Employee" identified by a field "personType". 

Can I define types "Contact" which fetches Person with filter (personType="contact") and  "Employee" which fetches Person with filter (personType="employee")? Whats the best way to do this? One option I thought was to define separate fetchers for each type (Person, Contact and Employee) that applies respective filters. But this would duplicate the fetcher code for each type. Hence wanted to see if there is a better way to achieve this.

**Comment by srinivasankavitha** [comment]
If I understand correctly, in this case you would define Person as a union type that could be either a Contact or Employee. Then you can define a data fetcher that returns this union type based on the filter
E.g. In your schema
```
type Query {
     person(filter: String) : Person
}

union Person = Contact | Employee
```
Then your data fetcher can be:

```
@DgsQuery
Person person(String filter) {
     if (filter == "Contact") {
     // fetch and return Contact
    } else  {
    // fetch and return Employee
   }
}
}
```


Hope this helps.

**Reply by mattkallo** [reply]
Thanks @srinivasankavitha for the response. In the above case, how are Contact and Employee defined? I am looking for something like below:

```

type Contact ** filtered Person by personType="contact" ** {
  name,
  email
}

type Employee ** filtered Person by personType="employee" ** {
  name,
  email
}

```

**Reply by srinivasankavitha** [reply]
Per my understanding I assume Contact and Employee are the different types, so the data fetcher would return the union type set to Person or Employee based on some filtering field.
